iosxr_test.py

```diff
@@ -234,44 +233,68 @@ def test_xr():
         logger.debug("Vagrant SSH to XR Console is sane")
         return True
 
+def cleanup_vagrant():
+    """Clean up after ourselves."""
+    logger.info("Cleaning up...")
 
-def main():
-    # Get virtualbox
-    global input_box
-    global verbose
+    run(['vagrant', 'destroy', '--force'], cont_on_error=True)
+
+    # Clean up Vagrantfile
+    try:
+        os.remove('Vagrantfile')
+    except OSError:
+        pass
 
+def parse_args():
+    """Parse the CLI arguments."""
     parser = argparse.ArgumentParser(description='Run basic unit-test on a Vagrant VirtualBox')
     parser.add_argument('BOX_FILE',
                         help='local Vagrant VirtualBox filename')
     parser.add_argument('-v', '--verbose',
                         action='store_const', const=logging.DEBUG,
                         default=logging.INFO, help='turn on verbose messages')
 
-    args = parser.parse_args()
-    verbose = args.verbose
+    return parser.parse_args()
+
+
+def main(vbox=None, verbosity=logging.INFO):
+    """Main test function."""
+    # Get virtualbox
+    global input_box
+    global verbose
+
+    if vbox is None:
+        args = parse_args()
+        verbose = args.verbose
+
+        input_box = args.BOX_FILE
+    else:
+        input_box = vbox
+        verbose = verbosity
 
-    input_box = args.BOX_FILE
     if not os.path.exists(input_box):
-        sys.exit(input_box, 'does not exist')
+        raise AbortScriptException('%s does not exist' % input_box)
 
-    logger.setLevel(level=args.verbose)
+    logger.setLevel(level=verbose)
 
-    # Bring the newly generated virtualbox up
-    bringup_vagrant()
+    try:
+        # Bring the newly generated virtualbox up
+        bringup_vagrant()
 
-    # Test IOS XR Linux
-    result_linux = test_linux()
+        # Test IOS XR Linux
```

Comment:
Ah - github shades the indent - which makes me think there's a hard tab inserted.